Source/WebCore/platform/Timer.h

```diff
@@ -66,24 +66,33 @@ class TimerBase {
     inline void stop();
     inline bool isActive() const;
 
-    MonotonicTime nextFireTime() const { return m_heapItem ? m_heapItem->time : MonotonicTime { }; }
+    MonotonicTime nextFireTime() const { return m_heapItemWithBitfields.pointer() ? m_heapItemWithBitfields.pointer()->time : MonotonicTime { }; }
     WEBCORE_EXPORT Seconds nextFireInterval() const;
     Seconds nextUnalignedFireInterval() const;
     Seconds repeatInterval() const { return m_repeatInterval; }
 
     void setTimerAlignment(TimerAlignment& alignment) { m_alignment = alignment; }
     TimerAlignment* timerAlignment() { return m_alignment.get(); }
 
-    bool hasReachedMaxNestingLevel() const { return m_hasReachedMaxNestingLevel; }
-    void setHasReachedMaxNestingLevel(bool value) { m_hasReachedMaxNestingLevel = value; }
+    bool hasReachedMaxNestingLevel() const { return bitfields().hasReachedMaxNestingLevel; }
+    void setHasReachedMaxNestingLevel(bool);
 
-    void augmentFireInterval(Seconds delta) { setNextFireTime(m_heapItem->time + delta); }
+    void augmentFireInterval(Seconds delta) { setNextFireTime(m_heapItemWithBitfields.pointer()->time + delta); }
     void augmentRepeatInterval(Seconds delta) { augmentFireInterval(delta); m_repeatInterval += delta; }
 
     void didChangeAlignmentInterval();
 
     WEBCORE_EXPORT static void fireTimersInNestedEventLoop();
 
+protected:
+    struct TimerBitfields {
+        uint16_t hasReachedMaxNestingLevel : 1 { false };
+        uint16_t shouldRestartWhenTimerFires : 1 { false }; // DeferrableOneShotTimer
```

Comment:
Why `uint16_t`? Why not `bool`, or `unsigned`, or `uint8_t`?

Reply:
So that sizeof(TimerBitfields) == sizeof(uint16_t).
Otherwise, bitwise_cast is going to complain about the size mismatch.
We could use uint8_t if you think that's better though.

Reply:
I think so. `uint16_t` seems mysterious, whereas `uint8_t` is obviously "smallest round number of bytes".